ctapipe_io_lst/__init__.py

```diff
@@ -281,19 +327,27 @@ def fill_lst_event_container_from_zfile(self, event):
     def fill_r0_camera_container_from_zfile(self, r0_container, event):
         """
         Fill with R0CameraContainer
-
         """
 
+        # look for correct trigger_time (TAI time in ns), first in UCTS and then in TIB
 
-        # temporary patch to have an event time set
-        r0_container.trigger_time = (
-            self.data.lst.tel[self.camera_config.telescope_id].evt.tib_pps_counter +
-            self.data.lst.tel[self.camera_config.telescope_id].evt.tib_tenMHz_counter * 10**(-7))
+        if self.data.lst.tel[self.camera_config.telescope_id].evt.ucts_timestamp > 0:
+            r0_container.trigger_time = self.data.lst.tel[self.camera_config.telescope_id].evt.ucts_timestamp
 
-        if r0_container.trigger_time is None:
+        elif self.data.lst.tel[self.camera_config.telescope_id].evt.tib_pps_counter >0:
+            r0_container.trigger_time = (
```

Comment:
...evt.ucts_timestamp is an absolute timestamp, on the contrary, if UCTS fails you define here trigger_time as the time that has elapsed since a starting ntp. Is that right? Either you define it as an absolute value or a value relative to a starting point.

Reply:
This is how I was taking the timing info in lstchain:

```
ucts_timestamp = event.lst.tel[telescope_id].evt.ucts_timestamp
 
if ucts_timestamp != 0:
    # UCTS timestamps are nanoseconds in UNIX timestamp format and must be given in seconds
    utc_time = Time(datetime.utcfromtimestamp(ucts_timestamp / 1e9))
else:
     start_ntp = event.lst.tel[telescope_id].svc.date
     ntp_time = start_ntp + event.r0.tel[telescope_id].trigger_time
     utc_time = Time(datetime.utcfromtimestamp(ntp_time))
```

Reply:
So it appears to me that it first checks if TIB info is present;  if so, then it will take TIB info and ignore other information. Therefore, it means TIB info will be given priority. If this is the case, then it is fine.